src/main/java/com/exasol/adapter/properties/TableCountLimit.java

```diff
@@ -0,0 +1,85 @@
+     */
+    public static PropertyValidator getValidator() {
+        return PropertyValidator.optional(MAXTABLES_PROPERTY, TableCountLimit::validatePropertyValue);
+    }
+
+    private static void validatePropertyValue(final String value) throws PropertyValidationException {
+        try {
+            final int parsed = Integer.parseUnsignedInt(value);
+            if (parsed == 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (final IllegalArgumentException exception) {
+            throw new PropertyValidationException(ExaError.messageBuilder("E-VSCJDBC-43") //
+                    .message("Invalid parameter value.") //
+                    .mitigation("The adapter property {{max_tables_property}}" //
+                            + " if present, must be a positive integer.") //
+                    .parameter("max_tables_property", MAXTABLES_PROPERTY) //
+                    .toString());
+        }
+    }
+
+    /**
+     * @param properties Adapter Properties passed to {@code CREATE VIRTUAL SCHEMA}
+     * @return new instance of {@link TableCountLimit} based on the properties
+     */
+    public static TableCountLimit from(final AdapterProperties properties) {
+        if (properties.containsKey(MAXTABLES_PROPERTY)) {
+            final int value = Integer.parseUnsignedInt(properties.get(MAXTABLES_PROPERTY));
+            return new TableCountLimit(value);
+        }
+        return new TableCountLimit(DEFAULT_MAX_MAPPED_TABLE_LIST_SIZE);
+    }
+
+    private final int maxNumberOfTables;
+
+    /**
+     * @param maxNumberOfTables maximum number of tables to be accepted
+     */
+    public TableCountLimit(final int maxNumberOfTables) {
+        this.maxNumberOfTables = maxNumberOfTables;
+    }
+
+    /**
+     * Verify that the given number of mapped tables does not exceed the configured max table limit.
+     *
+     * @param numberOfTables actual size of mapped tables
+     * @throws RemoteMetadataReaderException if the table limit has been exceeded
+     */
+    public void validateNumberOfTables(final int numberOfTables) throws RemoteMetadataReaderException {
+        if (numberOfTables > this.maxNumberOfTables) {
+            throw new RemoteMetadataReaderException(ExaError.messageBuilder("E-VSCJDBC-42")
+                    .message("The size of the list of the selected tables exceeds" //
+                            + " the configured allowed maximum of {{current_limit}}.")
+                    .mitigation(
+                            "Please use the {{table_filter_property}} property to define the list of tables you need" //
+                                    + " or increase the limit using the {{max_tables_property}} property.") //
+                    .parameter("current_limit", this.maxNumberOfTables) //
+                    .parameter("table_filter_property", TABLE_FILTER_PROPERTY) //
+                    .parameter("max_tables_property", MAXTABLES_PROPERTY) //
```

Comment:
No need to use parameters for constants. Then the message is easier to find in the error catalog. Just use string concatenation.